cromwell-drs-localizer/src/main/scala/drs/localizer/DrsLocalizerMain.scala

```diff
@@ -91,76 +142,86 @@ class DrsLocalizerMain(drsUrl: String,
     }
+  }
 
-    def maybeRetryForDownloadFailure(t: Throwable): IO[DownloadResult] = {
-      t match {
-        case _: FatalRetryDisposition =>
-          IO.raiseError(t)
-        case _ if downloadAttempt < downloadRetries =>
-          backoff foreach { b => Thread.sleep(b.backoffMillis) }
-          logger.warn(s"Attempting retry $downloadAttempt of $downloadRetries download retries to download $drsUrl", t)
-          resolveAndDownloadWithRetries(downloadRetries, checksumRetries, downloaderFactory, backoff map { _.next }, downloadAttempt + 1, checksumAttempt)
-        case _ =>
-          IO.raiseError(new RuntimeException(s"Exhausted $downloadRetries download retries to resolve, download and checksum $drsUrl", t))
-      }
+  def buildGoogleDownloaders(resolvedGoogleUrls: List[ResolvedDrsUrl]) : List[Downloader] = {
+    resolvedGoogleUrls.map{url=>
+      downloaderFactory.buildGcsUriDownloader(
+        gcsPath = url.drsResponse.gsUri.get,
+        serviceAccountJsonOption = url.drsResponse.googleServiceAccount.map(_.data.spaces2),
+        downloadLoc = url.downloadDestinationPath,
+        requesterPaysProjectOption = requesterPaysProjectIdOption)
     }
+  }
+  def buildBulkAccessUrlDownloader(resolvedUrls: List[ResolvedDrsUrl]) : Downloader = {
+    downloaderFactory.buildBulkAccessUrlDownloader(resolvedUrls)
+  }
 
-    resolveAndDownload(downloaderFactory).redeemWith({
-      maybeRetryForDownloadFailure
-    },
-    {
-      case f: FatalDownloadFailure =>
-        IO.raiseError(new RuntimeException(s"Fatal error downloading DRS object: $f"))
-      case r: RetryableDownloadFailure =>
-        maybeRetryForDownloadFailure(
-          new RuntimeException(s"Retryable download error: $r for $drsUrl on retry attempt $downloadAttempt of $downloadRetries") with RegularRetryDisposition)
-      case ChecksumFailure =>
-        maybeRetryForChecksumFailure(new RuntimeException(s"Checksum failure for $drsUrl on checksum retry attempt $checksumAttempt of $checksumRetries"))
-      case o => IO.pure(o)
-    })
+  /**
+    * Runs a synchronous HTTP request to resolve the provided DRS URL with the provided resolver.
+    */
+  def resolveSingleUrl(resolverObject: DrsLocalizerDrsPathResolver, drsUrlToResolve: UnresolvedDrsUrl): IO[ResolvedDrsUrl] = {
+    val fields = NonEmptyList.of(DrsResolverField.GsUri, DrsResolverField.GoogleServiceAccount, DrsResolverField.AccessUrl, DrsResolverField.Hashes)
+    val drsResponse = resolverObject.resolveDrs(drsUrlToResolve.drsUrl, fields)
+    drsResponse.map(resp => ResolvedDrsUrl(resp, drsUrlToResolve.downloadDestinationPath, toValidatedUriType(resp.accessUrl, resp.gsUri)))
   }
 
-  private [localizer] def resolveAndDownload(downloaderFactory: DownloaderFactory): IO[DownloadResult] = {
-    resolve(downloaderFactory) flatMap { _.download }
+
+  val defaultBackoff: CloudNioBackoff = CloudNioSimpleExponentialBackoff(
+    initialInterval = 10 seconds, maxInterval = 60 seconds, multiplier = 2)
```

Comment:
This seems aggressive (or, not aggressive enough?) - maybe start with 1 second interval? Is there a max number of retries set elsewhere?

Reply:
This is what we were using before, but happy to tone it down a bit. I will also put the max retry count variable here (I went with 5) so everything is in the same place.